src/lib.rs

```diff
@@ -430,35 +429,97 @@ fn expand_template(
     name: &ProjectName,
     dir: &Path,
     template_values: &HashMap<String, toml::Value>,
-    template_config: Option<Config>,
+    template_config: Config,
     args: &Args,
 ) -> Result<()> {
     let crate_type: CrateType = args.into();
-    let template = template::substitute(name, &crate_type, template_values, args.force)?;
-    let template = match template_config.as_ref() {
-        None => Ok(template),
-        Some(config) => {
-            project_variables::fill_project_variables(template, config, args.silent, |slot| {
-                interactive::variable(slot)
-            })
-        }
-    }?;
+    let template = template::create_liquid_object(name, &crate_type, template_values, args.force)?;
+    let template = project_variables::fill_project_variables(
+        template,
+        &template_config,
+        args.silent,
+        |slot| interactive::variable(slot),
+    )?;
+
+    let (template_cfg, template) = merge_conditionals(template_config, template, args)?;
+
     let mut pbar = progressbar::new();
 
-    ignore_me::remove_unneeded_files(dir, args.verbose);
+    ignore_me::remove_unneeded_files(dir, &template_cfg.ignore, args.verbose)?;
 
-    template::walk_dir(
-        dir,
-        template,
-        template_config.and_then(|c| c.template),
-        &mut pbar,
-    )?;
+    match template::walk_dir(dir, template, template_cfg, &mut pbar) {
+        Ok(_) => {}
+        Err(e) => {
+            let e = format! {"{}", e};
+            bail!(e);
+        }
+    }
```

Comment:
isn't `if let Err(e) = emplate::walk_dir(dir, template, template_cfg, &mut pbar)` better in this case?
Because you are ignoring the `Ok` branch

Reply:
I'm at work atm. but if my memory recalls it correctly, this piece might even be there due to temporary debugging needs - obviously needs a little love